### PR TITLE
feat: #6017 market: retrieval ask CLI command

### PR DESCRIPTION
**Pull request by dirkmc**
## Related Issues
Fixes https://github.com/filecoin-project/lotus/issues/6017
Supersedes https://github.com/filecoin-project/lotus/pull/7397

## Proposed Changes
Add CLI command for retrieval asks

## Example Output
```
$ lotus client query-retrieval-ask --size=134217728 f01000 bafyaeicumxtb5zs3fnli2yndp4iakfit64jdxp7ulajv1lkw2cyiwi3pvm
Ask: f01000
Unseal price: 0.000000001 FIL
Price per byte: 0.000000002 FIL
Payment interval: 1 MiB
Payment interval increase: 2 MiB
Total price for 134217728 bytes: 0.268435457 FIL
```

### CODE REVIEW

File: cli/client.go [hunk 1]
```diff
@@ -1030,6 +1031,63 @@ var clientFindCmd = &cli.Command{
 	},
 }
 
+var clientQueryRetrievalAskCmd = &cli.Command{
+	Name:      "query-retrieval-ask",
```
Comment (magik6k) [inline cli/client.go:1035]: Maybe just
```suggestion
	Name:      "retrieval-ask",
```

File: cli/client.go [hunk 2]
```diff
@@ -1030,6 +1031,63 @@ var clientFindCmd = &cli.Command{
...
+			return err
+		}
+
+		afmt.Printf("Ask: %s\n", maddr)
+		afmt.Printf("Unseal price: %s\n", types.FIL(ask.UnsealPrice))
+		afmt.Printf("Price per byte: %s\n", types.FIL(ask.PricePerByte))
+		afmt.Printf("Payment interval: %s\n", types.SizeStr(types.NewInt(ask.PaymentInterval)))
+		afmt.Printf("Payment interval increase: %s\n", types.SizeStr(types.NewInt(ask.PaymentIntervalIncrease)))
+
+		size := cctx.Int64("size")
```
Comment (magik6k) [inline cli/client.go:1079]: I'm pretty sure that QueryOffer has a Size field (for the whole piece, so doesn't exactly apply to selective retrievals, but imo still useful to print that)

**Review by nonsense — APPROVED** [review]
lgtm

File: cli/client.go (hunk 2 above)
Comment (dirkmc) [reply cli/client.go:1079]: The size is actually passed in as a parameter - if the user passes it, then the "Total Cost" calculation is output - see the Example Output in the PR description.

Comment (magik6k) [reply cli/client.go:1079]: Yes, my point is that the default could be to use the size from QueryOffer if it wasn't specified by the user

Comment (dirkmc) [reply cli/client.go]: Ok I made that change 👍

**Review by magik6k — APPROVED** [review]

**Pull request merged**